src/resources/object.rs

```diff
@@ -490,6 +494,8 @@ impl Object {
             .headers(crate::get_headers().await?)
             .send()
             .await?
+            .error_for_status()
+            .map_err(Error::from)?
```

Comment:
This explicit conversion is not needed, the ?-operator will do the conversion automatically.